packages/button/src/index.tsx

```diff
@@ -51,36 +88,65 @@ const ButtonFCFactory: <AdditionalProps extends {}>(
   as: StyledComponentElement
 ) => React.FC<ButtonProps> = <AdditionalProps extends {}>(
   as: StyledComponentElement = 'button'
-) => ({children, innerRef, ...props}) => {
-  // eslint-disable-next-line react-hooks/rules-of-hooks
-  const defaults = useDefaults('button', props, {
-    height: 32,
-    appearance: 'default' as ButtonAppearanceType,
-    intent: 'none' as ButtonIntentType,
-    isLoading: false,
-    full: false
-  })
+) => {
+  const ButtonFC = ({children, innerRef, ...props}) => {
+    const defaults = useDefaults('button', props, {
+      height: 32,
+      appearance: 'default' as ButtonAppearanceType,
+      intent: 'none' as ButtonIntentType,
+      isLoading: false,
+      full: false
+    })
+
+    const theme = useTheme()
+
+    const IconComponent = props.icon
+
+    return (
+      <StyledText
+        as={as}
+        borderRadius={getBorderRadiusForControlHeight(defaults.height)}
+        variant={getTextSizeForControlHeight(defaults.height)}
+        ref={innerRef}
+        {...defaults}
+        {...props}
+        disabled={props.disabled || props.isLoading}
+      >
+        {children}
+        {defaults.isLoading && (
+          <StyledSpinner
+            marginLeft={
```

Comment:
we shouldn't calculate styles inside `render()`'s JSX, instead put `iconPosition` as a prop and calculate margins inside `StyledSpinner`'s styles

Reply:
done